### PR TITLE
fix python syntax errors in docstring examples

### PR DESCRIPTION
**Pull request by ev-br**
The objective is to make sure the code in docstrings is valid python, not to have docstrings doctestable. 
While I'm at it, I tried to tweak the code snippets so that play ball with the online JS thingy which toggles the output and python REPL prompt, so that it's easy for a user to copy-paste the code into their interactive console. 

I used a simple extension of `tools/refguide_check.py` script, here https://gist.github.com/ev-br/69cee8f232d49c609ba2
[Edit: relevant code is in tools/refguide_check.py]

Most of the diff is trivial, with only a couple of points of note, which I highlight below for the maintainers of the relevant modules.

### CODE REVIEW

File: scipy/linalg/_decomp_update.pyx.in [hunk 1]
```diff
@@ -1718,15 +1718,15 @@ def qr_insert(Q, R, u, k, which='row', rcond=None, overwrite_qru=False, check_fi
...
+    ...               [  7.,   8.,  -6.]])
     >>> q, r = linalg.qr(a)
 
     Given this QR decomposition, update q and r when 2 rows are inserted.
                       
     >>> u = np.array([[  6.,  -9.,  -3.], 
-                      [ -3.,  10.,   1.]])
-    >>> q1, r1 = linalg.qr_insert(q, r, u, 2, 'row', False)
+    ...               [ -3.,  10.,   1.]])
+    >>> q1, r1 = linalg.qr_insert(q, r, u, 2, 'row')
```
Comment (ev-br) [inline scipy/linalg/_decomp_update.pyx.in]: `rcond=False` does not pass through the validation, which only lets None through.

File: scipy/optimize/_minimize.py [hunk 2]
```diff
@@ -314,10 +314,10 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
              Function evaluations: 64
              Gradient evaluations: 64
     >>> res.x
-    [ 1.  1.  1.  1.  1.]
-    >>> print res.message
+    array([ 1.  1.  1.  1.  1.])
+    >>> print(res.message)
     Optimization terminated successfully.
-    >>> res.hess
+    >>> res.hess_inv
```
Comment (ev-br) [inline scipy/optimize/_minimize.py]: This looks to be just a typo: `res` does not have a `hess` attribute, it is documented to have the inverse Hessian, and `hess_inv` attribute exists  (and what I get locally agrees with the values in the docstring).

File: scipy/optimize/nonlin.py [hunk 3]
```diff
@@ -1350,7 +1350,7 @@ class KrylovJacobian(Jacobian):
         preconditioners. For example,
 
         >>> jac = BroydenFirst()
-        >>> kjac = KrylovJacobian(inner_M=jac.inverse).
+        >>> kjac = KrylovJacobian(inner_M=jac.inverse)
```
Comment (ev-br) [inline scipy/optimize/nonlin.py]: I'm not sure what is this. `BroydenFirst().inverse` does not exist. Should it be `jac.aspreconditioner`?

File: scipy/linalg/basic.py [hunk 4]
```diff
@@ -551,7 +552,7 @@ def solve_circulant(c, b, singular='raise', tol=None,
 
     Check by solving one pair of `c` and `b` vectors (cf. ``x[1, 1, :]``):
 
-    >>> solve_circulant(c[1], b[:,1])
+    >>> solve_circulant(c[1], b[1, :])
```
Comment (ev-br) [inline scipy/linalg/basic.py]: This seems to be just a typo left over from API tweaks. The original version throws a ValueError, the tweaked version works and seems to make sense.

File: scipy/optimize/_minimize.py (hunk 2 above)
Comment (dlax) [inline scipy/optimize/_minimize.py]: Ack.

File: scipy/optimize/nonlin.py (hunk 3 above)
Comment (dlax) [inline scipy/optimize/nonlin.py]: Maybe `InverseJacobian(jac)` assuming the type of `inner_M` is correct.

**Comment by rgommers** [comment]
@ev-br nice idea, would be useful to add this to ``refguide_check.py`` as an option (in a separate PR).

**Comment by ev-br** [comment]
@rgommers OK, done (it's just a couple of lines, so it's easier to do than to remember to submit a separate PR)

File: scipy/optimize/nonlin.py (hunk 3 above)
Comment (rgommers) [inline scipy/optimize/nonlin.py]: No idea what was intended here, but the suggestion sounds reasonable. At least

 >>> kjac = optimize.nonlin.KrylovJacobian(inner_M=optimize.nonlin.InverseJacobian(jac))

doesn't give an error. This is not a public function, so maybe just make the change now and get this PR merged.

Comment (rgommers) [inline scipy/optimize/nonlin.py]: and you actually need to add this in order to not trigger ``refguide_check.py``:

    >>> from scipy.optimize.nonlin import BroydenFirst, KrylovJacobian, InverseJacobian

**Comment by rgommers** [comment]
This all looks good to me except for the one loose end in ``KrylovJacobian`` to finish up.

File: scipy/optimize/nonlin.py (hunk 3 above)
Comment (ev-br) [inline scipy/optimize/nonlin.py]: Done.

**Pull request merged**